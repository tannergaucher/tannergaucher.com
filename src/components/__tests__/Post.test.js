import React from 'react'
import { render } from 'react-testing-library'
import renderer from 'react-test-renderer'
import Post from '../Post'

describe(`<Post/>`, () => {
  const props = {
    title: 'hello world',
    date: '10.25.2018',
    slug: 'fake-post-slug',
    intro: 'fake intro sentence',
  }
  it(`renders and matches snapshot`, () => {
    const tree = renderer.create(<Post {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it(`renders a post title`, () => {
    const { getByText } = render(<Post {...props} />)
    const testTitle = getByText(props.title)
    expect(testTitle).toBeInTheDocument()
    expect(testTitle).toHaveTextContent(props.title)
  })
  it(`renders an intro sentence`, () => {
    const { getByText } = render(<Post {...props} />)
    const testIntro = getByText(props.intro)
    expect(testIntro).toBeInTheDocument()
    expect(testIntro).toHaveTextContent(props.intro)
  })
  it(`renders a post date`, () => {
    const { getByText } = render(<Post {...props} />)
    const testDate = getByText(props.date)
    expect(testDate).toBeInTheDocument()
    expect(testDate).toHaveTextContent(props.date)
  })
})
