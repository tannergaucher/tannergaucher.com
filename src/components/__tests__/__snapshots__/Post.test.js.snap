// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/> renders and matches snapshot 1`] = `
<MockedLink
  className="sc-jqCOkK qFArh"
  none="true"
  primary="true"
  to="fake-post-slug"
>
  <div
    className="sc-bbmXgH cJydhv sc-bwzfXH hFUbDu sc-bdVaJa iBHnqn"
  >
    <div
      className="sc-uJMKN eZyOLu sc-gqjmRU jmmWQb"
    >
      10.25.2018
    </div>
    <div
      className="sc-uJMKN bSMuRG sc-gqjmRU bmbyAD"
    >
      hello world
    </div>
    <div
      className="sc-gqjmRU kuSGKZ"
    >
      fake intro sentence
    </div>
  </div>
</MockedLink>
`;
